process/process_linux.go

```diff
@@ -894,3 +911,38 @@ func AllProcesses() (map[int32]*FilledProcess, error) {
 	}
 	return procs, nil
 }
+
+// ensurePathReadable ensures that the current user is able to read the path in question
+// before opening it.  on some systems, attempting to open a file that the user does
+// not have permission is problematic for customer security auditing
+func ensurePathReadable(path string) error {
+	euid := uint32(os.Geteuid())
```

Comment:
> Can we move this call up the call stack
Yes with a caveat

The agent just calls [`AllProcesses`](https://github.com/DataDog/gopsutil/blob/dd/process/process_linux.go#L840) where we could shim this in easily enough.  But it would require calling getuid/geteuid in the individual methods on the process like [`Process#Cwd`](https://github.com/DataDog/gopsutil/blob/dd/process/process_linux.go#L840) or [`Process#IOCounters`](https://github.com/DataDog/gopsutil/blob/dd/process/process_linux.go#L193)